framework/src/modules/interoperability/mainchain/commands/submit_mainchain_cross_chain_update.ts

```diff
@@ -114,6 +142,8 @@ export class SubmitMainchainCrossChainUpdateCommand extends BaseCrossChainUpdate
 				// would refer to an inbox where the message has not been appended yet).
 				await this.internalMethod.appendToInboxTree(context, params.sendingChainID, ccmBytes);
 			}
+		} catch (error) {
+			panic(context.logger, error as Error);
```

Comment:
it was failing on terminateChainInternal, so shouldnt it panic there?

Reply:
`apply` function is written in a way that it will not throw error but handles it, `terminateChain` and log event. Apply function doesn't expect `terminateChain` or `sendInternal` to throw at the time when we are applying. So throwing of `apply` is unusual behaviour and if we catch and panic would work in this case.